src/Components/Register/Register.tsx

```diff
@@ -1,7 +1,246 @@
+import { IRest } from "./ResetInterface/ResetInterface";
+
+
+export default function Register({ saveAdminData }) {
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<IRest>();
+  const [isLoading, setIsLoding] = useState(false);
+  const {baseUrl} = useContext(AuthContext);
+  const theme = useTheme();
+  // ****************** to register-Password **********************
+  const onSubmit = (data: IRest) => {
+    setIsLoding(true);
+
+    axios
+      .post(`${baseUrl}/admin/users`, data)
+
+      .then((response) => {
+        // console.log(response);
+        navigate("/login");
+        toast.success("Successfully");
+      })
+      .catch((error) => {
+        console.log(error);
+        toast.error(error);
+        setIsLoding(false);
+      });
+  };
   return (
-    <div>Register</div>
-  )
+    <>
+      <Stack spacing={30} direction="row" justifyContent="center">
+        <Box sx={{ display: "flex", flexDirection: "column" }}>
+          <CardContent sx={{ flex: "1 0 auto", marginLeft: 5 }}>
+            <Typography
+              sx={{ mb: 1 }}
+              component="div"
+              variant="h5"
+              className="text"
+            >
+              Stay<span className="text2">cation.</span>
+            </Typography>
+            <Box sx={{ marginLeft: 5, }}>
+              <Typography variant="h3" component="div" sx={{ mb: 2 }}>
+              Sign up
+              </Typography>
+              <Typography variant="h6" component="div" className="text3">
+                If you don’t have an account register <br />
+                You can <span>Login here !</span>
+              </Typography>
+              <FormControl
+                onSubmit={handleSubmit(onSubmit)}
+                component="form"
+                sx={{
+                  "& .MuiTextField-root": {
+                    mt: 4,
+                    width: "35ch",
+                    height: "5ch",
+                  },
+                }}
+                noValidate
+                autoComplete="off"
+              >
+                <div>
+                  <TextField
+                    label="User Name"
+                    type="text"
+                    id="outlined-size-normal"
+                    placeholder="Please type here ..."
+                    color="primary"
+                    focused
+                    {...register("userName", {
+                      required: true,
+                     
+                    })}
+                  />
+                  <Box sx={{ color: "red", pt: 1 }}>
+                    {errors.userName && errors.userName.type === "required" && (
+                      <p>User Name is required</p>
+                    )}
+                  </Box>
+                
+                </div>
+                <div>
+                  <TextField
+                    label="Phone Number"
+                    type="number"
```

Comment:
Phone Number should be of type string not a number, since they are allowed to have a leading zero, - , + and spaces. and these letter are not allowed in normal numbers.